### PR TITLE
Added ability to pass a console object to Cmd.print_to().

### PR DESCRIPTION
**Pull request by kmvanbrunt**
See #1578

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,10 @@
+# 3.2.3 (TBD)
+
+- Enhancements
+    - Added ability to pass a console object to `Cmd.print_to()`. This provides support for things
+      like wrapping a `print_to()` call in a `console.status()` or `console.capture()` context
+      manager.
```
Comment (tleonhardt) [inline CHANGELOG.md:6]: We should document that the `file` parameter of `cmd2.Cmd.print_to` has been renamed to `destination` and that this is a potentially breaking change.

As such we should probably make it a 3.3.0 release instead of 3.2.3.

File: cmd2/cmd2.py [hunk 2]
```diff
@@ -1259,13 +1264,30 @@ def print_to(
...
+        """Print objects to a given destination (file stream or cmd2 console).
 
         This method is configured for general-purpose printing. By default, it enables
         soft wrap and disables Rich's automatic detection for markup, emoji, and highlighting.
         These defaults can be overridden by passing explicit keyword arguments.
 
-        :param file: file stream being written to
+        See the Rich documentation for more details on emoji codes, markup tags, and highlighting.
+
+        !!! note
```
Comment (tleonhardt) [inline cmd2/cmd2.py:1293]: Oh wow, I didn't realize it would render things like notes correctly within docstrings.  This is great. I tested it locally and verified it works as intended.

**Review by tleonhardt — CHANGES_REQUESTED** [review]

File: cmd2/cmd2.py [hunk 3]
```diff
@@ -1247,30 +1252,66 @@ def visible_prompt(self) -> str:
...
+        destination: IO[str] | Cmd2BaseConsole,
         *objects: Any,
         sep: str = " ",
         end: str = "\n",
         style: StyleType | None = None,
-        soft_wrap: bool = True,
-        emoji: bool = False,
-        markup: bool = False,
-        highlight: bool = False,
+        soft_wrap: bool | None = None,
```
Comment (tleonhardt) [inline cmd2/cmd2.py:1260]: Should we add more unit tests for output methods now that these parameters can also be `None` instead of a `bool`?

File: CHANGELOG.md (hunk 1 above)
Comment (kmvanbrunt) [reply CHANGELOG.md:6]: Done

File: cmd2/cmd2.py (hunk 3 above)
Comment (kmvanbrunt) [reply cmd2/cmd2.py:1260]: Since `Cmd2GeneralConsole` already sets these values to `False`, changing the arg defaults to `None` doesn't affect existing behavior.

**Review by tleonhardt — APPROVED** [review]
